### ISSUE TITLE
[terra-form] Select options to support nullable option

### ISSUE THREAD

**Issue by dnguyenchevy** [cerner/terra-core#651]
# Issue Description
Select component defaults to the first option as selected. We have a use case within our form where the select options are optional and would prefer to default to a null option.

e.g.
```
<label htmlFor="selectFruit">What's your favorite fruit?</label>
<select id="selectFruit" name="fruit" defaultValue="">
  <option value="">Select a fruit</select>
  <option value="apple">Apple</option>
  <option value="oranges">Orange</option>
  <option value="banana">banana</option>
</select>
```

## Issue Type
- [ ] New Feature
- [x] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior
Provide a placeholder text for nullable option to be selected first, refer to example above.

## Current Behavior
The first choice provided to `Select` component is selected option.

**Comment by dnguyenchevy** [comment]
With incoming changes from https://github.com/cerner/terra-core/issues/538, it was determine we could just pass in an object with empty value with the placeholder text for display. 

This issue can be closed once #538 is completed.

**Comment by emilyrohrbough** [comment]
The changes for #538 were released in terra-forms v1.2.0.
